services/src/pro/contexts/mod.rs

```diff
@@ -120,6 +121,32 @@ where
         // as plots do not produce a stream of results, we have nothing to count for now
         op
     }
+
+    fn read_ml_model(&self, path: std::path::PathBuf) -> geoengine_operators::util::Result<String> {
+        let cfg = get_config_element::<crate::util::config::DataProvider>()
+            .map_err(|_| geoengine_operators::error::Error::InvalidDataProviderConfig)?;
+
+        let model_path = cfg.pro_ml_xgboost_model_path;
+
+        let model = std::fs::read_to_string(model_path.join(path))?;
+
+        Ok(model)
+    }
+
+    fn write_ml_model(
+        &mut self,
+        path: std::path::PathBuf,
+        ml_model_str: String,
+    ) -> geoengine_operators::util::Result<()> {
+        let cfg = get_config_element::<crate::util::config::DataProvider>()
+            .map_err(|_| geoengine_operators::error::Error::InvalidDataProviderConfig)?;
+
+        let model_path = cfg.pro_ml_xgboost_model_path;
+
+        std::fs::write(model_path.join(path), ml_model_str)?;
```

Comment:
please use the canionicalize method to ensure the  path doesn't escape the model_path base. This should prevent bad things if `path` is something like `../../`